design/design-doc-stacks.md

```diff
@@ -206,28 +206,31 @@ Inside of a package, the following filesystem layout is expected:
 
 ```
 .registry/
-├── icon.jpg
+├── icon.png
 ├── app.yaml # Application metadata.
 ├── install.yaml # Optional install metadata.
 ├── rbac.yaml # Optional RBAC permissions.
 └── resources
       └── databases.foocompany.io # Group directory
             ├── group.yaml # Optional Group metadata
-            ├── icon.jpg # Optional Group icon
+            ├── icon.png # Optional Group icon
             └── mysql # Kind directory by convention
                 ├── v1alpha1
                 │   ├── mysql.v1alpha1.crd.yaml # Required CRD
-                │   ├── icon.jpg # Optional resource icon
+                │   ├── icon.png # Optional resource icon
                 │   └── resource.yaml # Resource level metadata.
                 └── v1beta1
                     ├── mysql.v1beta1.crd.yaml
-                    ├── icon.jpg
+                    ├── icon.png
                     └── resource.yaml
 ```
 
 * `app.yaml`: This file is the general metadata and information about the Stack, such as its name, description, version, owners, etc.  This metadata will be saved in the `Extension` record's spec fields.
 * `install.yaml`: This file contains the information for how the custom controller for the Stack should be installed into Crossplane.  Initially, only simple `Deployment` based controllers will be supported, but eventually other types of implementations will be supported as well, e.g., templates, functions/hooks, templates, a new DSL, etc.
-* `resources` directory: This directory contains all the CRDs and optional metadata about them.  These CRDs are the types that the custom controller implements the logic for.  They will be directly installed into Crossplane so that users can create instances of them to start consuming their new Stack functionality.
+* `icon.png`: This file (or `icon.svg`, `icon.jpg`, `icon.gif`, or potentially other supported filenames, TBD) will be used in a visual context when listing or describing this stack.  The preffered formats/filenames are `svg`, `png`, `jpg`, `gif` in that order (if multiple files exist).  For bitmap formats, the width to height ratio should be 1:1. Limitations may be placed on the acceptable file dimensions and byte size (TBD).
```

Comment:
Is there value added in supporting more than just `svg` and `png`? Just more curious than anything else, and additionally I think `svg` should definitely be the preferred format for scaling purposes.

Reply:
I don't think anything outside of SVG and PNG should be necessary. It may be better to stay entirely agnostic, which I think could only reasonably be accomplished by having the `app.yaml` specify the location (and perhaps media type) of the icon(s).

Reply:
This is actually already suggested via https://github.com/crossplaneio/crossplane/pull/604/files#diff-5db872f8c3bbf22fd1da26bc1a5c1f30R140-R142, although I think I prefer a filename to a base64 value.  We could use URIs (and data URIs) to permit both.

Between `.yaml` attributes and magic filenames, we should pick one approach or the other, not recommend both.

Reply:
After chatting with @rathpc, I realize the URI approach is problematic.
 
I was implying `file:///stack-local/path`, but that does seem awkward and most likely would warrant abuse as `http://my/icon.png` (we don't want to reach out to the internet for images, I think)..

Removing the `file://` URI scheme and host is what I was getting at, using a stack file-system path relative to the `.yaml` file referencing the image.

Just looking for `icon.X`, `icon.Y`, `icon.Z` does seem simpler, for now at least, which suggests removing `icon` from the `app.yaml` file.

Reply:
Removed in b4f825b